### PR TITLE
Feature/support additional status codes

### LINKED ISSUE #3: support for additional status codes and scenarios
Follow up of https://github.com/microsoft/kiota/issues/1699 for Python.
No generation changes required (hence the issue being created over here). Only the runtime rules need to be implemented/validated. (see the linked PRs for examples).

### PR DESCRIPTION
**Pull request by samwelkanda**
Fixes #3 
Fixes https://github.com/microsoftgraph/msgraph-sdk-python/issues/448

### CODE REVIEW

File: kiota_http/httpx_request_adapter.py [hunk 1]
```diff
@@ -189,7 +200,7 @@ async def send_collection_async(
         self,
         request_info: RequestInformation,
         parsable_factory: ParsableFactory,
-        error_map: Dict[str, ParsableFactory],
+        error_map: Dict[str, ParsableFactory] = {},
```
Comment (baywet) [inline kiota_http/httpx_request_adapter.py:203]: would we need to update the interface as well if we made this optional?
also why would we make that change only for this method?

**Review by baywet — CHANGES_REQUESTED** [review]

File: kiota_http/httpx_request_adapter.py (hunk 1 above)
Comment (samwelkanda) [reply kiota_http/httpx_request_adapter.py]: Good catch. This was for local testing purposes and meant to be removed

Comment (samwelkanda) [reply kiota_http/httpx_request_adapter.py]: And no, setting a default value does not require an update to the interface. It's merely an extension

**Review by baywet — APPROVED** [review]
Thanks for making the change

**Pull request merged**